sway/input/seatop_down.c

```diff
@@ -110,3 +110,23 @@ void seatop_begin_down(struct sway_seat *seat, struct sway_container *con,
 	container_raise_floating(con);
 	transaction_commit_dirty();
 }
+
+void seatop_begin_down_on_layer_surface(struct sway_seat *seat,
+		struct wlr_surface *surface, uint32_t time_msec, double sx, double sy) {
+	seatop_end(seat);
+
+	struct seatop_down_event *e =
+		calloc(1, sizeof(struct seatop_down_event));
+	if (!e) {
+		return;
+	}
+	e->con = NULL;
```

Comment:
We only need ```e->con``` now for ```handle_unref```. Would it be feasible to compare ```e->surface == con->view->surface``` there as well or is the surface too volatile?

Reply:
We should stop the seatop_down on surface unmap, it doesn't make sense to keep it going if the surface is not visible anymore. So maybe we can just add an unmap listener?

Reply:
Unfortunately ```wlr_surface``` doesn't emit an unmap event. Only thing close would be destroy .. ?

Reply:
Hm right, I suppose destroy is fine then... It's not that of a big deal, it just means the user will maybe need to release the pointer button for the events to be sent if the surface underneath is unmapped.